src/modules/frequency/application/dtos/class-attendance.dto.ts

```diff
@@ -0,0 +1,150 @@
+import { ApiProperty } from "@nestjs/swagger";
+import { Transform, Type } from "class-transformer";
+import {
+  IsArray,
+  IsDate,
+  IsEnum,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  ValidateNested,
+} from "class-validator";
+import { FrequencyStatus } from "src/common/enums/domain.enums";
+export class StudentClassAttendanceItemDTO {
+  @ApiProperty({
+    example: 11,
+    description: "The unique ID of the frequency record"
+  })
+  frequencyId: number;
+
+  @ApiProperty({
+    example: 1,
+    description: "The ID of the student"
+  })
+  studentId: number;
+
+  @ApiProperty({
+    example: "João Silva",
+    description: "The student's full name"
+  })
+  studentFullName: string;
+
+  @ApiProperty({
+    example: "90.00%",
+    description: "The student's attendance percentage for this class"
+  })
+  attendancePercentage: string;
```

Comment:
I believe this should be a number, everywhere in the codebase, and we just add the % symbol in the frontend where needed. To simplify calculations and even styling (i.e. red/yellow/green tint)
```suggestion
  attendancePercentage: number;
```

Reply:
Ok. Aceita a sugestão.